cc/pass_alive_test.cc

```diff
@@ -0,0 +1,732 @@
+
+      {// Top-left region is pass-alive because only one empty point is not
+       // adjacent to the enclosing chain.
+       // board state
+       R"(O..X.....
+          ...X.....
+          XXXX.....
+          .X.X.....
+          XXXX.....
+          .........
+          .........
+          .........
+          .........)",
+       // expected result
+       R"(XXXX.....
+          XXXX.....
+          XXXX.....
+          XXXX.....
+          XXXX.....
+          .........
+          .........
+          .........
+          .........)"},
+
+      {// board state
+       R"(.........
+          .........
+          .........
+          .........
+          .........
+          .........
+          .........
+          OOOOO....
+          O.O.O....)",
+       // expected result
+       R"(.........
+          .........
+          .........
+          .........
+          .........
+          .........
+          .........
+          OOOOO....
+          OOOOO....)"},
+
+      {// board state
+       R"(.........
+          ..XXXXXX.
+          ..X..X.X.
+          ..XXXXXX.
+          .........
+          .........
+          .........
+          .........
+          .........)",
+       // expected result
+       R"(.........
+          ..XXXXXX.
+          ..XXXXXX.
+          ..XXXXXX.
+          .........
+          .........
+          .........
+          .........
+          .........)"},
+
+      {// board state
+       R"(.........
+          .OOOOOOO.
+          .O.OXXXO.
+          .OOOXXXO.
+          .OX.XXXO.
+          .OXXXXOO.
+          .OOOOOO..
+          .........
+          .........)",
+       // expected result
+       R"(.........
+          .OOOOOOO.
+          .OOOOOOO.
+          .OOOOOOO.
+          .OOOOOOO.
+          .OOOOOOO.
+          .OOOOOO..
+          .........
+          .........)"},
+
+      {// board state
+       R"(.........
+          .........
+          .........
+          .........
+          .........
+          .........
+          OOOOO....
+          .O.XOO...
+          O.O.XO...)",
+       // expected result
+       R"(.........
+          .........
+          .........
+          .........
+          .........
+          .........
+          OOOOO....
+          OOOOOO...
+          OOOOOO...)"},
+
+      {// board state
+       R"(OXX.X.X.O
+          OX.XOXXXX
+          OXXOOOOOO
+          OOOO.....
+          .........
+          .........
+          .........
+          .........
+          .........)",
+       // expected result
+       R"(OXXXXXXXX
+          OXXXOXXXX
+          OXXOOOOOO
+          OOOO.....
+          .........
+          .........
+          .........
+          .........)"},
+
+      {// board state
+       R"(..O.O.XX.
+          .OOOO.X.X
+          OO.....XX
+          .........
+          .........
+          X......OO
+          .XX...O.O
+          X.X..OO.O
+          .X.X.O.O.)",
+       // expected result
+       R"(..O.O.XXX
```

Comment:
Why is upper left not pass alive? As a human it is

Reply:
Top-left point is not a liberty of the white group, which means that white region is "large".  (there exist "human pass-alive" groups that do not meet Benson's definition of pass-alive)

Reply:
Oh hm, i'm not quite correct.  IIUC, the top left region has a non-adjacent point AND the group has only one vital region.  The difference between this and the one you commented on above (line 160) is that the group on line 160 has two vital regions already.   Hope i'm right this time :)

Reply:
Is there a polynomial time algorithm that returns the definitive status of groups?

Reply:
Black can capture by surrounding the white group then finally playing at `a`:

```
.XOaOX.
XOOOOX.
OOXXX..
XX.....
.......
```

Reply:
Benson's paper mentions an N! algorithm if that's any help :P